packages/adminrouter/extra/src/README.md

```diff
@@ -31,6 +31,230 @@ Use `make check-api-docs` to validate that the YAML and HTML files are up to dat
+ * EE agent specific code
+ * EE master specific code
+ * Open agent specific code
+ * Open master specific code
+on top of that, Nginx-specific configuration is divided into three sections:
+ * main
+ * HTTP
+ * server
+This gives us in total 27 possible "buckets" for Nginx directives. The
+differentiation for sections can be avoided if we decide to use some more
+advanced templating but this would further complicate configuration and
+make it more difficult for people to test and develop AR on live clusters.
+
+Directly from it, stems the idea how the NGINX includes can be structured:
+* open:
+```
+includes
+├── http
+│   ├── agent.conf
+│   ├── common.conf
+│   ├── master.conf
+│   └── open
+│       ├── common.conf
+│       └── master.conf
+├── main
+│   ├── common.conf
+│   └── open
+│       └── common.conf
+├── server
+│   ├── common.conf
+│   ├── master.conf
+│   └── open
+│       ├── agent.conf
+│       └── master.conf
+├── snakeoil.crt
+└── snakeoil.key
+```
+ * ee:
+```
+includes
+├── http
+│   └── ee
+│       ├── common.conf
+│       └── master.conf
+├── main
+│   └── ee
+│       └── common.conf
+└── server
+    └── ee
+        ├── agent.conf
+        ├── common.conf
+        └── master.conf
+```
+All Nginx related configuration directives reside in includes directory which
+contains directories reflecting all the sections present in NGINX configuration
+(http|main|server). Because they are always present, no
+matter the flavour/server type, they are chosen as top-level directories
+in the `includes` dir. The `common.conf`, `agent.conf` and `master.conf`
+files are flavour-agnostic thus they reside in each section's main dir
+and are present/included only in the Open AR repository. EE repo re-uses them
+after being applied on top of the Open repository. Contents of each of the
+files are as follows:
+ * `common.conf` contains all the things that are common across all flavours
+   and server types for given section
+ * `agent.conf` contains all the things that are common for all agents across
+   all flavours
+ * `master.conf` contains all the things that are common for all masters across
+   all flavours
+Each section can have either `ee` or `open` directories, but never both. `ee`
+directory is only present in EE repo, Open repo contains only `open` directories.
+Contents of these directories may be as follows:
+ * `common.conf` contains all the things that are common for given flavour, no
+   matter the server type
+ * `agent.conf` contains all the things that are common for agents in given
+   flavour
+ * `master.conf` contains all the things that are common for masters in given
+   flavour
+The order of includes is for the time being hard-coded in nginx.\*.conf files,
+but it does not matter that much as:
+ * [nginx by default orders the imports while globbing](https://serverfault.com/questions/361134/nginx-includes-config-files-not-in-order)
+ * the order of includes should not matter within a particular section. This means
```

Comment:
Reworded to:
```
~ 135 The order of includes is for the time being hard-coded in nginx.\*.conf files.
~ 136 Within particular section though it does not matter that much as:
  137  * [nginx by default orders the imports while globbing](https://serverfault.com/questions/361134/nginx-includes-config-files-not-in-order)
~_138  * location blocks *MUST NOT* rely on the order in which they appear
  139    in the config file and define matching rules precise enough to avoid ambiguity.
```